features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/WindowInspector.kt

```diff
@@ -10,54 +10,56 @@ import android.annotation.SuppressLint
     private val GLOBAL_WM_CLASS by lazy {
-        try {
-            return@lazy Class.forName("android.view.WindowManagerGlobal")
-        } catch (e: Throwable) {
-            // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
-            // sdkLogger.errorWithTelemetry("WindowInspector failed to retrieve the decor views", e)
-        }
-        return@lazy null
+        return@lazy Class.forName("android.view.WindowManagerGlobal")
     }
 
-    @Suppress("SwallowedException", "TooGenericExceptionCaught")
+    @Suppress("SwallowedException", "TooGenericExceptionCaught", "UnsafeThirdPartyFunctionCall")
     private val GLOBAL_WM_INSTANCE by lazy {
-        try {
-            return@lazy GLOBAL_WM_CLASS?.getMethod("getInstance")?.invoke(null)
-        } catch (e: Throwable) {
-            // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
-            // sdkLogger.errorWithTelemetry("WindowInspector failed to retrieve the decor views", e)
-        }
-        return@lazy null
+        return@lazy GLOBAL_WM_CLASS?.getMethod("getInstance")?.invoke(null)
     }
 
-    @Suppress("SwallowedException", "TooGenericExceptionCaught")
+    @Suppress("SwallowedException", "TooGenericExceptionCaught", "UnsafeThirdPartyFunctionCall")
     private val VIEWS_FIELD by lazy {
-        try {
-            return@lazy GLOBAL_WM_CLASS?.getDeclaredField("mViews")
-        } catch (e: Throwable) {
-            // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
-            // sdkLogger.errorWithTelemetry("WindowInspector failed to retrieve the decor views", e)
-        }
-        return@lazy null
+        return@lazy GLOBAL_WM_CLASS?.getDeclaredField("mViews")
     }
 
-    fun getGlobalWindowViews(): List<View> {
-        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-            WindowInspector.getGlobalWindowViews()
-        } else {
-            getGlobalWindowViewsLegacy(GLOBAL_WM_INSTANCE, VIEWS_FIELD)
+    @SuppressWarnings("TooGenericExceptionCaught")
+    fun getGlobalWindowViews(internalLogger: InternalLogger): List<View> {
+        return try {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                WindowInspector.getGlobalWindowViews()
+            } else {
+                getGlobalWindowViewsLegacy(internalLogger, GLOBAL_WM_INSTANCE, VIEWS_FIELD)
+            }
+        } catch (e: Throwable) {
+            internalLogger.log(
+                InternalLogger.Level.ERROR,
+                InternalLogger.Target.TELEMETRY,
+                { FAILED_TO_RETRIEVE_DECOR_VIEWS_ERROR_MESSAGE },
+                e,
+                true
+            )
+            emptyList()
         }
     }
 
     @Suppress("SwallowedException", "TooGenericExceptionCaught")
-    internal fun getGlobalWindowViewsLegacy(globalWmInstance: Any?, viewsField: Field?): List<View> {
+    internal fun getGlobalWindowViewsLegacy(
```

Comment:
Same here, we should not catch a generic exception if we can. Only exceptions (no pun intended) would be if we called a customer's callback (and even then, we should only catch `Exception`  and not `Throwable`)

Reply:
same as above